specification/schemaregistry/SchemaRegistry/main.tsp

```diff
@@ -0,0 +1,265 @@
+    @body
+    schemaContent: bytes;
+  },
+  SchemaProperties
+>;
+
+//  PUT https://{namespaceName}.servicebus.windows.net/$schemaGroups/{groupName}/schemas/{name}?api-version={apiVersion}
+#suppress "@azure-tools/typespec-azure-core/use-standard-operations" "Non-standard operation."
+#suppress "@azure-tools/typespec-azure-core/byos" "storage service is not used"
+@summary("Register new schema")
+@doc("Register new schema. If schema of specified name does not exist in specified group, schema is created at version 1. If schema of specified name exists already in specified group, schema is created at latest version + 1.")
+@route("/$schemaGroups/{groupName}/schemas/{name}")
+@put
+op registerSchema is Azure.Core.Foundations.Operation<
+  {
+    @doc("Name of schema group.")
+    @path
+    groupName: string;
+
+    @doc("Name of schema.")
+    @path
+    name: string;
+
+    ...SchemaContent;
+    ...SchemaContentType;
+  },
+  NoContentResponse<SchemaProperties>
+>;
+
+// Models
+alias NoContentResponse<T> = TypeSpec.Http.Response<204> & T;
+
+union SchemaContentTypeValues {
+  customBinary: "application/octet-stream",
+  custom: "text/plain; charset=utf-8",
+  avro: "application/json; serialization=Avro",
+  json: "application/json; serialization=Json",
+  protobuf: "text/vnd.ms.protobuf",
+}
+
+alias SchemaContentType = {
+  @doc("The content type for given schema.")
+  @header("Content-Type")
+  contentType: SchemaContentTypeValues;
+};
+
+alias SchemaContent = {
+  @body
+  @doc("String representation (UTF-8) of the schema.")
+  content: bytes;
+};
+
+@doc("Metadata of a schema.")
+model SchemaProperties {
+  @doc("URL location of schema, identified by schema group, schema name, and version.")
+  @header("Location")
+  location: string;
+
+  @doc("References specific schema in registry namespace.")
+  @header("Schema-Id")
+  schemaId: string;
+
+  @doc("URL location of schema, identified by schema ID.")
+  @header("Schema-Id-Location")
+  schemaIdLocation: string;
+
+  @doc("References schema group.")
+  @header("Schema-Group-Name")
+  schemaGroupName: string;
+
+  @doc("References schema name.")
+  @header("Schema-Name")
+  schemaName: string;
+
+  @doc("Version of the returned schema.")
+  @header("Schema-Version")
+  schemaVersion: int32;
+}
+
+@doc("The schema, including its metadata and content.")
+model Schema {
+  ...SchemaProperties;
+  ...SchemaContent;
+  ...SchemaContentType;
+  ...OkResponse;
+}
+
+/**
+ * The list of strings with server paging support.
+ */
+@Azure.Core.pagedResult
+model StringsList {
+  /** The collection of pageable items. */
+  @Azure.Core.items
+  value: string[];
+
+  /** The link to the next page of items */
+  @Azure.Core.nextLink
+  nextLink?: string;
+}
+
+/**
+ * The list of int32s with server paging support.
+ */
+@Azure.Core.pagedResult
+model Int32sList {
+  /** The collection of pageable items. */
+  @Azure.Core.items
+  value: int32[];
```

Comment:
This should match what the service is returning

Reply:
The service is just returning the list of ints, and not a list of key/int values.

Reply:
so I believe this should be right